instaclustr/resource_cluster.go

```diff
@@ -630,7 +630,7 @@ func resourceClusterRead(d *schema.ResourceData, meta interface{}) error{
 	d.Set("cluster_name", cluster.ClusterName)
 
 	clusterProvider := make(map[string]interface{}, 0)
-	mapstructure.Decode(cluster.Provider, &clusterProvider)
+	mapstructure.Decode(cluster.Provider[0], &clusterProvider)
```

Comment:
**[Question]** Why would we support multiple providers if it's only the first that is being used?

Reply:
Will need to update this to use all of the providers from the list not just the first one

Reply:
Talked with @Chrismelba-Instaclustr on this, we will keep this as is since the TF provider won't support multiple providers in it's current state.

Reply:
As discussed, console supports multi-provider provisioning, but API currently doesn't. This structure allows the API to return the information for a multi-provider cluster, but terraform will just take the first instance returned since it doesn't support multi-provider provisioning (thus there will only be one provider returned in the list)